### PR TITLE
[Workflow] The TransitionEvent is able to modify the context

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        |

EUFOSSA

ping @HeahDude

### CODE REVIEW

File: src/Symfony/Component/Workflow/Workflow.php [hunk 1]
```diff
@@ -308,17 +308,20 @@ private function leave($subject, Transition $transition, Marking $marking): void
         }
     }
 
-    private function transition($subject, Transition $transition, Marking $marking): void
+    private function transition($subject, Transition $transition, Marking $marking, array $context): array
```
Comment (vudaltsov) [inline src/Symfony/Component/Workflow/Workflow.php]: Why not passing as a reference?

**Review by vudaltsov — APPROVED** [review]

File: src/Symfony/Component/Workflow/Workflow.php (hunk 1 above)
Comment (lyrixx) [reply src/Symfony/Component/Workflow/Workflow.php]: I don't really know. We don't use reference too often in the Symfony world.

What do other people think about that?

Comment (HeahDude) [reply src/Symfony/Component/Workflow/Workflow.php]: The event won't deal with a reference so I don't really see the point.

**Review by HeahDude — APPROVED** [review]
LGTM

**Review by xabbuh — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @lyrixx.

**Pull request merged**